lib/vquic/curl_osslq.c

```diff
@@ -1461,24 +1469,71 @@ static CURLcode cf_osslq_check_and_unblock(struct Curl_cfilter *cf,
 {
   struct cf_osslq_ctx *ctx = cf->ctx;
   struct h3_stream_ctx *stream;
+  size_t poll_count = 0;
+  size_t result_count = 0;
+  size_t idx_count = 0;
+  CURLcode res = CURLE_OK;
+  struct timeval timeout;
 
   if(ctx->h3.conn) {
     struct Curl_llist_node *e;
+
+    res = CURLE_OUT_OF_MEMORY;
+
+    if(ctx->item_count < Curl_llist_count(&data->multi->process)) {
+      ctx->item_count = 0;
+      ctx->poll_items = realloc(ctx->poll_items,
+                                Curl_llist_count(&data->multi->process) *
+                                sizeof(SSL_POLL_ITEM));
+      if(!ctx->poll_items)
```

Comment:
The `ctx→poll_items` buffer leaks if `realloc()` fails and returns `NULL`.
Same for `ctx->curl_items` below.

Reply:
how does it leak? Both poll_items and curl_items are freed in cf_osslq_ctx_free.  If these leak, then stream_bufcp, streams, and peer all must leak as well (working under the assumption that in some cases cf_osslq_ctx_free must not get called unilaterally on shutdown).

Reply:
I see, thank you, will fix